src/cargo/core/resolver/resolve.rs

```diff
@@ -80,11 +80,9 @@ pub enum ResolveVersion {
     /// V3 by default staring in 1.53.
     #[default]
```

Comment:
See <https://github.com/rust-lang/cargo/issues/12120#issuecomment-1607753875>

I'll post another PR for doc update.

For MSRV-aware lockfile generating, see <https://github.com/rust-lang/cargo/pull/12861>.

Reply:
At the *latest*, we might want to default to the new format for 2024-edition crates, since those crates won't build with old cargo anyway so there won't be any "flapping" between formats for different Cargo versions.

Reply:
Further more, it does sound like we might want to check if `edition` and `rust-version` are in sync and don't conflict with each other.

Reply:
We should have checks that the rust-version isn't older than the edition allows.

Falling back to an edition check if the rust-version isn't set would be useful.  The problem is we don't have a workspace edition field.  We could do something like `rust-version` and check the oldest edition present.